client/my-sites/design-preview/index.js

```diff
@@ -0,0 +1,182 @@
+
+const DesignPreview = React.createClass( {
+
+	propTypes: {
+		// Any additional classNames to set on this wrapper
+		className: React.PropTypes.string,
+		// True to show the preview; same as WebPreview.
+		showPreview: React.PropTypes.bool,
+		// The viewport device to show initially; same as WebPreview but defaults to 'tablet'.
+		defaultViewportDevice: React.PropTypes.string,
+		// Show close button; same as WebPreview.
+		showClose: React.PropTypes.bool,
+		// Elements to render on the right side of the toolbar; same as WebPreview.
+		children: React.PropTypes.node,
+		// A function to run when the preview has loaded. Will be passed a ref to the iframe document object.
+		onLoad: React.PropTypes.func,
+		// These are all provided by the state
+		previewMarkup: React.PropTypes.string,
+		customizations: React.PropTypes.object,
+		actions: React.PropTypes.object,
+		isUnsaved: React.PropTypes.bool,
+		selectedSiteId: React.PropTypes.number,
+	},
+
+	getDefaultProps() {
+		return {
+			showPreview: false,
+			defaultViewportDevice: 'tablet',
+			showClose: true,
+			customizations: {},
+			actions: {},
+			isUnsaved: false,
+			onLoad: noop,
+		};
+	},
+
+	componentDidMount() {
+		this.loadPreview();
+	},
+
+	componentDidUpdate( prevProps ) {
+		// If there is no markup or the site has changed, fetch it
+		if ( ! this.props.previewMarkup || this.props.selectedSiteId !== prevProps.selectedSiteId ) {
+			this.loadPreview();
+		}
+		// Refresh the preview when it is being shown (since this component is
+		// always present but not always visible, this is similar to loading the
+		// preview when mounting).
+		if ( this.props.showPreview && ! prevProps.showPreview ) {
+			this.loadPreview();
+		}
+		// If the customizations have been removed, restore the original markup
+		if ( this.haveCustomizationsBeenRemoved( prevProps ) ) {
+			// Force the initial markup to be restored because the DOM may have been
+			// changed, and simply not applying customizations will not restore it.
+			debug( 'restoring original markup' );
+			this.loadPreview();
+		}
+		// Apply customizations
+		if ( this.props.customizations && this.previewDocument ) {
+			debug( 'updating preview with customizations', this.props.customizations );
+			updatePreviewWithChanges( this.previewDocument, this.props.customizations );
+		}
+	},
+
+	haveCustomizationsBeenRemoved( prevProps ) {
+		return ( this.props.previewMarkup &&
+			this.props.customizations &&
+			this.props.previewMarkup === prevProps.previewMarkup &&
+			prevProps.customizations &&
+			Object.keys( this.props.customizations ).length === 0 &&
+			Object.keys( prevProps.customizations ).length > 0
+		);
+	},
+
+	loadPreview() {
+		if ( this.props.selectedSiteId && this.props.actions.fetchPreviewMarkup ) {
+			debug( 'loading preview with customizations', this.props.customizations );
+			this.props.actions.fetchPreviewMarkup( this.props.selectedSiteId, '', this.props.customizations );
```

Comment:
> 1. The my-sites area for a specific site first loads.
2. The preview is made visible (in case there are changes).
3. When customizations had been applied but have since been removed.

> an example refreshing the settings page, showing the network request on page load…

Doesn't this conform to the expected behaviour? I believe we want to load the preview on the initial page load, and for every site switched. We could switch it to fetch on demand, but we'd lose the 'instant' factor, which the editor preview has.